readme.md

```diff
@@ -1,62 +1,70 @@
-# Dog Search 
+# Image Search Application with Weaviate
 
 This project's origin is [here](https://github.com/weaviate/weaviate-examples/tree/main/nearest-neighbor-dog-search).
 
 ## Description
-In this demo, we will use Weaviate to create an image-search application. We'll develop a dog image-search application that lets users upload a photo of a dog and receive a list of dog breeds that are the most similar.
+This guide demonstrates how to build an Image Search Application with Weaviate. The application allows users to upload images of dogs and find the most similar dog breed. While we focus on dogs, you can adapt this approach for various image recognition and product search applications.
 
-(TODO: Add demo video)
+### Used Technologies
+- Weaviate
+- Flask
+- Python
+- Docker
 
-## Used Technologies
-Weaviate Vector DB
-Python Programming Language (v3.10+)
-Weaviate Python Client Library
-Flask (For WebApp interface development)
-Docker Desktop
+**Weaviate Modules:**
+- img2vec-neural
 
 ## Prerequisites
-Docker Desktop should be installed and running.
-Python 3.10 or later should be installed.
+Before starting, ensure you have the following:
 
-## Setup instructions
-Note: We recommend you create a new virtual environment for this.
+1. Basic knowledge of vector search.
+2. A Weaviate instance set up using Docker. For installation, refer to [Weaviate Installation](https://github.com/semi-technologies/weaviate).
+3. Docker installed on your system. Get it from [Docker Installation](https://www.docker.com/).
+
+## Setup Instructions
+1. Clone this repository to your local machine.
+2. Navigate to the `nearest-neighbor-dog-search` directory.
+3. Start the Weaviate instance using Docker:
+   ```bash
+   docker compose up -d
+   ```
+4. Verify Weaviate's status:
+   ```bash
+   python weaviate-test.py
+   ```
+   You should see an output similar to: `{"classes": []}`.
+
+## Usage Instructions
+### Image Vectorization
+1. Add your images to the `flask-app/static/img` folder.
+2. Execute `images-to-base64.py` to convert the images to base64 encoding:
    ```bash
-   python -m venv .venv
-   source .venv/bin/activate
+   python images-to-base64.py
    ```
+   The base64-encoded images will be stored in the `base64_images` folder.
 
-1. Install the Python packages.
+### Weaviate Database
+1. Define the schema in `create-schema.py` to structure the data.
+2. Add the schema to Weaviate:
+   ```bash
+   python create-schema.py
+   ```
+3. Upload data objects in batches using `upload-data-objects.py`:
    ```bash
-    pip install -r requirements.txt
```

Comment:
Please also keep the installation of required packages in the setup instructions